Code/src/main/java/nl/utwente/group10/haskell/type/ConstT.java

```diff
@@ -1,35 +1,73 @@
 package nl.utwente.group10.haskell.type;
 
+import com.google.common.base.Joiner;
+
 /**
- * A specific, non-variable Haskell type.
+ * Constant, concrete type. However, it may consist of variable types.
  */
 public class ConstT extends Type {
     /**
-     * The Haskell name of this type.
+     * The constructor for this type.
+     */
+    private final String constructor;
+
+    /**
+     * The types of the arguments for this type.
+     */
+    private final Type[] args;
+
+    /**
+     * @param constructor The constructor for this constant type.
+     * @param args The types of the arguments that this type accepts.
```

Comment:
Might want to add '...if any' - quite a lot of types have nullary type constructors.